pytorch_lightning/trainer/training_tricks.py

```diff
@@ -80,3 +100,223 @@ def configure_accumulated_gradients(self, accumulate_grad_batches):
+
+        factor: value which the old batch size is multiplied by to get the
+            new batch size
+
+        value: if a value is given, will override the batch size with this value.
+            Note that the value of `factor` will not have an effect in this case
+
+        desc: either `succeeded` or `failed`. Used purely for logging
+
+    """
+    model = trainer.get_model()
+    batch_size = getattr(model.hparams, batch_arg_name)
+    if value:
+        setattr(model.hparams, batch_arg_name, value)
+        new_size = value
+        if desc:
+            log.info(f'Batch size {batch_size} {desc}, trying batch size {new_size}')
+    else:
+        new_size = int(batch_size * factor)
+        if desc:
+            log.info(f'Batch size {batch_size} {desc}, trying batch size {new_size}')
+        setattr(model.hparams, batch_arg_name, new_size)
+    return new_size
+
+
+def _run_power_scaling(trainer, model, new_size, batch_arg_name, max_trials):
+    """ Batch scaling mode where the size is doubled at each iteration until an
+        OOM error is encountered. """
+    count = 0
+    while True:
+        garbage_collection_cuda()
+        trainer.global_step = 0  # reset after each try
+        try:
+            # Try fit
+            trainer.fit(model)
+            count += 1
+            if count > max_trials:
+                break
+            # Double in size
+            new_size = _adjust_batch_size(trainer, batch_arg_name, factor=2.0, desc='succeeded')
+        except RuntimeError as exception:
+            # Only these errors should trigger an adjustment
+            if is_oom_error(exception):
+                # If we fail in power mode, half the size and return
+                garbage_collection_cuda()
+                new_size = _adjust_batch_size(trainer, batch_arg_name, factor=0.5, desc='failed')
+                break
+            else:
+                raise  # some other error not memory related
+    return new_size
+
+
+def _run_binsearch_scaling(trainer, model, new_size, batch_arg_name, max_trials):
+    """ Batch scaling mode where the size is initially is doubled at each iteration
+        until an OOM error is encountered. Hereafter, the batch size is further
+        refined using a binary search """
+    high = None
+    count = 0
+    while True:
+        garbage_collection_cuda()
+        trainer.global_step = 0  # reset after each try
+        try:
+            # Try fit
+            trainer.fit(model)
+            count += 1
+            if count > max_trials:
+                break
+            # Double in size
+            low = new_size
+            if high:
+                if high - low <= 1:
+                    break
+                midval = (high + low) // 2
+                new_size = _adjust_batch_size(trainer, batch_arg_name, value=midval, desc='succeeded')
+            else:
+                new_size = _adjust_batch_size(trainer, batch_arg_name, factor=2.0, desc='succeeded')
```

Comment:
I would rather move this to the else section as here we do not expect any failed, right?

Reply:
this is meant to check if we are still in the initial phase of doubling the batch size or we have failed once (i.e. high is defined) and thus is in the binary search phase

Reply:
Sure, I mean
```
try:
  Do something 
except:
  Do if the something failed 
else:
  Do others if the something pass
```